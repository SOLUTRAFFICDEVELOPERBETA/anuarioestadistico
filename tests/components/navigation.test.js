import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import Router from 'next/router';
import { RouterContext } from 'next/dist/next-server/lib/router-context'
import Navegacion from '../../components/navigation';
import AuthContext from '../../contexts/auth';
import { user } from '../fixtures';

describe('Pruebas en <Navegacion/>', () => {
    const router = {
        pathname: '/',
        route: '',
        basePatch: '',
        query: {},
        asPath: '',
        push: jest.fn(),
        replace: jest.fn(),
        reload: jest.fn(),
        back: jest.fn(),
        prefetch: jest.fn(),
        beforePopState: jest.fn(),
        listen: jest.fn(),
        isFallback: false,
        events: {
            on: jest.fn(),
            off: jest.fn(),
            emit: jest.fn(),
        },

    }
    Router.router = router;
    const onLogOut = jest.fn();
    const wrapper = mount(
        <AuthContext.Provider value={{
            auth: 't0qnOa2EpoOcPeZ1qCYs0lypggi1',
            user,
            onLogOut

        }}>
            <RouterContext.Provider value={Router}>
                <Navegacion />
            </RouterContext.Provider>
        </AuthContext.Provider>

    )

    jest.mock('next/link', () => {
        return ({ children }) => {
            return children;
        }
    })
    test('Debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe mostrar el nombre correctamente', () => {
        expect(wrapper.find('a').at(0).text().trim()).toBe(user.name)
    })

    test('debe funcionar el renderizada /', () => {
        // wrapper.find('a').at(1).prop('onClick')({
        //     preventDefault(){},
        //     currentTarget: {
        //         nodeName: ''
        //     }
        // })

    })

    test('debe llamar la funci贸n de cerrar sesi贸n', () => {
        wrapper.find('img').prop('onClick')()

        expect(wrapper.find('img').prop('src')).toBe('/static/icons/salida.svg');
        expect(onLogOut).toHaveBeenCalled();
        expect(onLogOut).toHaveBeenCalledTimes(1);

    })

    test('debe mostrar el bot贸n de inicio de sesi贸n', () => {
        const wrapper = mount(
            <AuthContext.Provider value={{
                auth: '',
                user,
                onLogOut

            }}>
                <RouterContext.Provider value={router}>
                    <Navegacion />
                </RouterContext.Provider>
            </AuthContext.Provider>

        )
        expect(wrapper.find('img').prop('src')).toBe('/static/icons/login.svg');
    })
})