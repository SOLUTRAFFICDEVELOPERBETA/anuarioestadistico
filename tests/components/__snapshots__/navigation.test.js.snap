// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Navegacion/> Debe mostrarse correctamente 1`] = `
<nav
  className="css-1wt3c2j-Nav e19ybrxq1"
>
  <a
    className="css-6sh57k-NavLink e19ybrxq2"
  >
    Jose manuel Prueba
  </a>
  <a
    className="css-6sh57k-NavLink e19ybrxq2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Inicio
  </a>
  <a
    className="css-6sh57k-NavLink e19ybrxq2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Nosotros
  </a>
  <a
    className="css-6sh57k-NavLink e19ybrxq2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Páginas
  </a>
  <a
    className="css-6sh57k-NavLink e19ybrxq2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Configuración
  </a>
  <img
    className="css-10vqwh5-LogoOut e19ybrxq0"
    onClick={[Function]}
    src="/static/icons/salida.svg"
  />
</nav>
`;
